packages/flame/test/components/component_test.dart

```diff
@@ -652,6 +652,55 @@ void main() {
       );
     });
 
+    group('Replacing components', () {
```

Comment:
since we are reusing the remove and add methods, other side-effects and consequences and lifecycle statuses are already tested individually